### PR TITLE
ref(ui): Move PipelineView to new webpack entry

### PR DESCRIPTION
**Pull request by billyvg**
Refactor our bootstrap script to more discrete modules. Add a separate webpack entry for rendering pipeline view. Move initial rendering of SPA/PipelineView out of server templates and into the JS bundles. This is necessary for splitting up frontend deploys as there are more guarantees of modules being loaded inside of the JS entry, instead of in the server templates.

Note, I've removed the router instrumentation for Pipeline view as it significantly increases bundle size due to importing `app/routes`.

The Pipeline entrypoint also does not include `bootstrap/js/*` imports.

Works locally:
image

### CODE REVIEW

**Comment by billyvg** [comment]
Will test on staging before deploying

**Review by scefali — COMMENTED** [review]
As far as the React pipeline views go, this looks good to me

**Review by scefali — APPROVED** [review]
Approved for staging

File: src/sentry/static/sentry/app/bootstrap/renderOnDomReady.tsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export function renderOnDomReady(renderFn: () => void) {
+  if (document.readyState === 'complete') {
+    renderFn();
+  } else {
+    document.addEventListener('DOMContentLoaded', renderFn);
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/bootstrap/renderOnDomReady.tsx:5]: Heck yeah no more jquery

File: src/sentry/static/sentry/app/bootstrap/initializeMain.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import routes from 'app/routes';
+import ConfigStore from 'app/stores/configStore';
+import {metric} from 'app/utils/analytics';
+
+import {initializeSdk} from './initializeSdk';
+import {renderMain} from './renderMain';
+import {renderOnDomReady} from './renderOnDomReady';
+
+export function initializeMain() {
+  // SDK INIT  --------------------------------------------------------
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/bootstrap/initializeMain.tsx:16]: Do these big banner comments really add anything?

File: src/sentry/static/sentry/app/bootstrap/initializeSdk.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+ * (e.g.  `app/views/integrationPipeline`)
+ */
+function getSentryIntegrations(hasReplays: boolean = false, routes?: Function) {
+  const integrations = [
+    new ExtraErrorData({
+      // 6 is arbitrary, seems like a nice number
+      depth: 6,
+    }),
+    new Integrations.BrowserTracing({
+      ...(typeof routes === 'function'
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/bootstrap/initializeSdk.tsx:24]: Can `routingInstrumentation` just be set to `undefined`? Might be slightly cleaner

```tsx
new Integrations.BrowserTracing({
	routingInstrumentation: routes !== 'function'
		? undefined
		: Sentry.reactRouterV3Instrumentation(
	        Router.browserHistory as any,
	        Router.createRoutes(routes()),
	        Router.match
	      )
});
```

Comment (billyvg) [reply src/sentry/static/sentry/app/bootstrap/initializeSdk.tsx:24]: It can't

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/bootstrap/initializeSdk.tsx:24]: 😞

**Review by evanpurkhiser — APPROVED** [review]
Looks reasonable

File: src/sentry/static/sentry/app/bootstrap/initializeMain.tsx (hunk 2 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/bootstrap/initializeMain.tsx:16]: Will remove 👍

**Comment by billyvg** [comment]
@EvanPurkhiser ran into some problems with these `__SENTRY__xxxxxx` globals when testing on staging - so refactored to remove them, and instead pass configuration about to bootstrap scripts:  https://github.com/getsentry/sentry/pull/25174/files/bc2214916179031054652ed36d61ecfbd5806f46..5b09887dfe551bb37f14f171a2a4d4791f2073db

**Comment by billyvg** [comment]
Staging works with the latest changes.

**Pull request merged**